### DISCUSSION TITLE
Need Help Opening Server (from Visual Studio Code)

### DISCUSSION THREAD

**Discussion by jackiebgregory** [nss-evening-cohort-14/evening-client-side#13] [Help Me!] [answered]
Name: Jackie 

Have you searched the other help tickets? 
Yes, there is a related/similar question about the hosting server default, but that didn't help me with this question.

Describe your issue:
I'm watching the recording of last night and following along with the Star Wars project. When I get to the step of opening the server I can't seem to get it to work (yet).

List any error messages that appear:

➜  week1-starwars git:(main) hs -o
zsh: command not found: serve

Then I tried sudo hs -o and got this:

week1-starwars git:(main) sudo hs -o
events.js:292
      throw er; // Unhandled 'error' event
      ^
Error: listen EADDRINUSE: address already in use 0.0.0.0:8080
    at Server.setupListenHandle [as _listen2] (net.js:1318:16)
    at listenInCluster (net.js:1366:12)
    at doListen (net.js:1503:7)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
Emitted 'error' event on Server instance at:
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 8080
}
➜  week1-starwars git:(main) 

What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links)
I've been searching, but I clearly don't understand what I should be searching for based on the results I'm getting. I've tried "how to open server from vscode," "hs -o in vscode," "connecting to server in vscode," but I am not on the right track here.

What have you tried? Tell us everything you have tried.
I've tried clearing it and retyping it, I tried sudo hs -o, and I've tried googling it. Does this have something to do with the SSH key? 

Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with:
https://github.com/jackiebgregory/week1-starwars

Additional Note: When I was coding along last night, I had the same problem, but then it did work after using sudo. I was able to see the blue background then. However, after that, after making any updates in the code, they did not go through and show up on the blue background.

**Comment by drteresavasquez** [comment]
You need to make sure that every time you stop your server you use the command and not close the tab out. When you start the server it tells you what to use. If you are on a mac it is CTRL-C to stop the server. 

For now, you should be able to run `http-server -o -p 8081`

The issue here is that you have a server already running and you need to stop it. We will have to walk you through it, but make sure to take notes because after we fix it, we want you to come back here and give the details on how it was fixed.

**Comment by drteresavasquez** [comment]
Also, here are the docs for http-server: https://www.npmjs.com/package/http-server

**Comment by drteresavasquez** [comment, marked as answer]
To check what is running on your port run `sudo lsof -i :8080`

You will see something that looks like this:

```
COMMAND  PID         USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
node             27649    drteresavasquez   30u  IPv4 0xd310771867986089      0t0  TCP *:http-alt (LISTEN)
node             27649    drteresavasquez   31u  IPv4 0xd310771868610f29      0t0  TCP localhost:http-alt->localhost:59334 (ESTABLISHED)
node             27649 drteresavasquez   32u  IPv4 0xd310771868a04909      0t0  TCP localhost:http-alt->localhost:59335 (ESTABLISHED)
```

To kill the process, use `kill -9 {PASTE-YOUR-PID}`

**Comment by drteresavasquez** [comment]
@jackiebgregory at your earliest convenience, please update this discussion with what you did that worked for you.

**Comment by jackiebgregory** [comment]
I used http-server -o -p 8081 as Dr. T posted above, and that worked. When I was finished, I used CTRL-C to stop it.
